server/plugin/utils.go

```diff
@@ -245,21 +245,24 @@ func (p *Plugin) ParseSubscriptionsToCommandResponse(subscriptionsList []*serial
 	return sb.String()
 }
 
-func (p *Plugin) GetOffsetAndLimitFromQueryParams(r *http.Request) (int, int, error) {
-	queryParamOffset := r.URL.Query().Get(constants.QueryParamOffset)
-	queryParamLimit := r.URL.Query().Get(constants.QueryParamLimit)
-	offset := 0
-	limit := 10
-	if queryParamOffset != "" && queryParamLimit != "" {
-		var err error
-		if offset, err = strconv.Atoi(queryParamOffset); err != nil {
-			return offset, limit, err
-		}
+func (p *Plugin) GetOffsetAndLimitFromQueryParams(r *http.Request) (offset, limit int) {
+	query := r.URL.Query()
+	var page int
+	if val, err := strconv.Atoi(query.Get(constants.QueryParamPage)); err != nil || val < 0 {
+		p.API.LogError(constants.InvalidPaginationQueryParam, "Error", err.Error())
+		page = constants.DefaultPage
+	} else {
+		page = val
+	}
 
-		if limit, err = strconv.Atoi(queryParamLimit); err != nil {
-			return offset, limit, err
-		}
+	val, err := strconv.Atoi(query.Get(constants.QueryParamPerPage))
+	switch {
+	case err != nil || val < 0 || val > constants.DefaultPerPageLimit: // We can keep max limit per page and default limit per page same
```

Comment:
Why are we keeping the default and max value same? This introduces a limitation in the API. And if we have to keep these values same, then we should keep the value high like "50".

Reply:
increased it to `50`

we can keep the default and max limit the same because I think keeping them separate doesn't make sense 
let's say if web app doesn't put this query-param then the server should return a feasible set of data up to a point it thinks would be feasible and if they put a higher limit then also our server should send up to that feasible number only

so in case of missing or invalid query param, we should send the best feasible data set only which can be send as max limit